src/coreclr/jit/abi.h

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#pragma once
+
+class ABIPassingSegment
+{
+    regNumber m_register    = REG_NA;
+    unsigned  m_stackOffset = 0;
+
+public:
+    bool IsPassedInRegister() const;
+    bool IsPassedOnStack() const;
```

Comment:
Is it relevant to track the other common qualifications like `HFA` and `HVA`?

Reply:
I don't think so. HFA's/HVA's are just passed in more registers than other struct arguments, so hopefully by the end of this clean up that's a detail that's constrained fully to be within the ABI classification and not leaked out anywhere to the rest of the JIT.